lib/internal/modules/esm/translators.js

```diff
@@ -443,9 +444,11 @@ translators.set('json', function jsonStrategy(url, source) {
   debug(`Loading JSONModule ${url}`);
   const pathname = StringPrototypeStartsWith(url, 'file:') ?
     fileURLToPath(url) : null;
+  const shouldCheckAndPopulateCJSModuleCache =
```

Comment:
Can you add a comment to here to why we checkfor `?` and `#`?

Reply:
```suggestion
  const shouldCheckAndPopulateCJSModuleCache =
    // We want to involve the CJS loader cache only for file: URL with no search query and no hash.
```